skimage/filters/rank/_percentile.py

```diff
@@ -38,11 +38,11 @@ def _apply(func, image, selem, out, mask, shift_x, shift_y, p0, p1,
            out_dtype=None):
 
     assert_nD(image, 2)
-    image, selem, out, mask, max_bin = _handle_input(image, selem, out, mask,
-                                                     out_dtype)
+    image, selem, out, mask, n_bins = _handle_input(image, selem, out, mask,
```

Comment:
Why this rename of `max_bin`?  It makes it hard to find the actual change in this patch.

Reply:
@stefanv, I totally feel your pain on not wanting to review a 200 line PR. I'll revert the rename of `max_bin`, but I think the name is misleading. I split this PR into two commits. You may want to look at the first commit in isolation:
https://github.com/scikit-image/scikit-image/pull/3501/commits/d396fe6ff71637d2ee21d096482dcf5e094c8409

The second commit is the rename, and I'm happy to revert it. Looking at the first one may make you see why `max_bin` isn't actually the variable of interest. The variable that is being used to iterate over is `n_bins`.

Reply:
I should rephrase myself, I'll revert the rename if you insist.

Reply:
Thanks, this is hugely helpful.

Reply:
glad I was able to help!